### PR TITLE
[MPOM-429] Support develop with JDK 21 (palantirJavaFormat update)

### PR DESCRIPTION
**Pull request by tisonkun**
See also https://github.com/diffplug/spotless/issues/1819#issuecomment-1726986315.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1168,12 +1168,14 @@ under the License.
         <plugin>
           <groupId>com.diffplug.spotless</groupId>
           <artifactId>spotless-maven-plugin</artifactId>
-          <version>2.37.0</version>
+          <version>2.39.0</version>
```
Comment (slawekjaranowski) [inline pom.xml:1171]: done in #134

File: pom.xml [hunk 2]
```diff
@@ -1168,12 +1168,14 @@ under the License.
...
           <configuration>
             <java>
               <!-- orders of used formatters are important MPOM-376 -->
               <!-- eg. palantir override importOrder, so should be first -->
-              <palantirJavaFormat />
+              <palantirJavaFormat>
+                <version>2.36.0</version>
+              </palantirJavaFormat>
```
Comment (slawekjaranowski) [inline pom.xml:1178]: please add link to spotless issue https://github.com/diffplug/spotless/issues/1774

**Comment by slawekjaranowski** [comment]
I would like to wait for fix in spotless: https://github.com/diffplug/spotless/pull/1822

If no before next release of parents we can consider it

**Review by slachiewicz — APPROVED** [review]

**Review by slawekjaranowski — APPROVED** [review]

**Comment by tisonkun** [comment]
Thanks for your review and suggestion!

We can still use this patch now as https://github.com/diffplug/spotless/pull/1822 is unresolved yet.

**Comment by slachiewicz** [comment]
Yes, I think patching spotless may take some more time.

**Comment by tisonkun** [comment]
May I ask the release plan including this patch? I create this patch due to my project want to upgrade to be compatible with JDK 21 dev :D

**Comment by slawekjaranowski** [comment]
@tisonkun  - Just out of curiosity - how this project impact on your project?
It is only used by Apache Maven projects as our parents.

**Comment by tisonkun** [comment]
Oops. It's not hawkeye where I can pin dependencies by myself and doesn't inherit the MPOM...

I forget where I come from now 🤣

**Comment by tisonkun** [comment]
@slawekjaranowski I encountered it again and remembered. I met this issue from Apache Curator who inherited this pom.

**Pull request merged**